src/lua/digest.lua

```diff
@@ -267,4 +311,74 @@ m['aes256cbc'] = {
     end
 }
 
+local function check_unsigned(value)
+    local value_type = type(value)
+    if value_type == 'number' then
+        if value >= 0 and value < 2^53 and math.floor(value) == value then
+            return value
+        end
+    elseif value_type == 'cdata' and ffi.istype('uint64_t', value) then
+        return value
+    end
+end
+
+for _, var in ipairs({'32', '64'}) do
+    local xxHash
+
+    local update_fn_name = 'XXH' .. var .. '_update'
+    local digest_fn_name = 'XXH' .. var .. '_digest'
+    local reset_fn_name = 'XXH' .. var .. '_reset'
+    local copy_fn_name = 'XXH' .. var .. '_copyState'
+    local methods = {
+        update = function(self, str)
+            if type(str) ~= 'string' then
+                error("Usage xxhash" .. var .. ":update(string)")
+            end
+            return ffi.C[update_fn_name](self, str, #str) == ffi.C.XXH_OK
```

Comment:
It makes sense to cache `ffi.C.<name>`. To avoid 2 index operations. Although I see it is not done in the other places, so maybe we could do it in a separate patch later or before this one. Up to you.

Reply:
According LuaJIT FFI manual it's wrong approach - https://luajit.org/ext_ffi_tutorial.html . See "To Cache or Not to Cache
" section. It's better to cache "ffi.C" than "ffi.C.<name>".

Reply:
Consider https://github.com/tarantool/tarantool/pull/5828/commits/0d546adae6f7727589c67c202e09409666324e9a

Reply:
Nice, didn't know this. Might explain some issues I have in my own patch in another ticket.